cmd/kanikoExecute.go

```diff
@@ -134,117 +135,222 @@ func runKanikoExecute(config *kanikoExecuteOptions, telemetryData *telemetry.Cus
+				commonPipelineEnvironment.container.imageNames = append(commonPipelineEnvironment.container.imageNames, entry.ContainerImageName)
+
+			case entry.ContainerImage != "":
+				containerImageName, err := docker.ContainerImageNameFromImage(entry.ContainerImage)
+				if err != nil {
+					log.SetErrorCategory(log.ErrorConfiguration)
+					return errors.Wrapf(err, "invalid name part in image %v", entry.ContainerImage)
 				}
-				if config.CreateBOM {
-					//Syft for multi image, generates bom-docker-(1/2/3).xml
-					return syft.GenerateSBOM(config.SyftDownloadURL, "/kaniko/.docker", execRunner, fileUtils, httpClient, commonPipelineEnvironment.container.registryURL, commonPipelineEnvironment.container.imageNameTags)
+				containerImageNameTag, err := docker.ContainerImageNameTagFromImage(entry.ContainerImage)
+				if err != nil {
+					log.SetErrorCategory(log.ErrorConfiguration)
+					return errors.Wrapf(err, "invalid tag part in image %v", entry.ContainerImage)
 				}
-				return nil
-			} else {
-				commonPipelineEnvironment.container.imageNames = append(commonPipelineEnvironment.container.imageNames, config.ContainerImageName)
-				commonPipelineEnvironment.container.imageNameTags = append(commonPipelineEnvironment.container.imageNameTags, fmt.Sprintf("%v:%v", config.ContainerImageName, containerImageTag))
-			}
 
-			log.Entry().Debugf("Single image build for image name '%v'", config.ContainerImageName)
-			containerImageNameAndTag := fmt.Sprintf("%v:%v", config.ContainerImageName, containerImageTag)
-			dest = []string{"--destination", fmt.Sprintf("%v/%v", containerRegistry, containerImageNameAndTag)}
-			commonPipelineEnvironment.container.imageNameTag = containerImageNameAndTag
-		} else if len(config.ContainerImage) > 0 {
-			log.Entry().Debugf("Single image build for image '%v'", config.ContainerImage)
-			containerRegistry, err := docker.ContainerRegistryFromImage(config.ContainerImage)
-			if err != nil {
-				log.SetErrorCategory(log.ErrorConfiguration)
-				return errors.Wrapf(err, "invalid registry part in image %v", config.ContainerImage)
+				log.Entry().Debugf("image build for image name '%v'", containerImageName)
+
+				buildOptions := append(config.BuildOptions,
+					"--context-sub-path", entry.ContextSubPath,
+					"--destination", entry.ContainerImage,
+				)
+				if err = runKaniko(config.DockerfilePath, buildOptions, config.ReadImageDigest, execRunner, fileUtils, commonPipelineEnvironment); err != nil {
+					return fmt.Errorf("failed to build image '%v' using '%v': %w", containerImageName, config.DockerfilePath, err)
+				}
+
+				commonPipelineEnvironment.container.imageNameTags = append(commonPipelineEnvironment.container.imageNameTags, containerImageNameTag)
+				commonPipelineEnvironment.container.imageNames = append(commonPipelineEnvironment.container.imageNames, containerImageName)
+			default:
+				return fmt.Errorf("either entry.ContainerImageName or entry.ContainerImage must be filled")
```

Comment:
Users can see this error, they don't know what is `entry.`

Reply:
refactored all error messages in multipleImages case, use `multipleImages:` prefix now